### ISSUE TITLE
What is a contribution?

### ISSUE THREAD

**Issue by Haislich** [KRLGroup/RL_2024#2]
# Problem

In the course grade breakdown, it is mentioned that contributing to a GitHub repository will result in a 5% bonus on the final grade.

![image](https://github.com/user-attachments/assets/7cf908a7-12a9-47a5-91c3-1861ceb0cffa)

However is never specified what a "contribution" is and how someone should make a valid contribution.

This course has students from different faculties, they might not have experience with GitHub or even know what contributing looks like. Without clear guidance, some students might miss out on this bonus just because they've never worked with GitHub.

# Solution

To make the bonus more accessible for everyone, I suggest to create a contributing section under the form of a `contributing.md` file, so it can be easily accessed/rendered from the web. The contributing section should provide a basic step-by-step guide on forking, making changes, and submitting pull requests. This must be done without assuming prior knowledge and keep things easy to follow for beginners.

This should allow all students to have fair shot at the 5% bonus.

**Comment by rikifunt** [comment]
You are correct: we do not require github knowledge to make a valid contribution. We wrote "to be agreed with the TAs" to mean that the contents and means to contribute *must* be discussed with the TAs before making the contribution, to make sure it is valid for the bonus, and to determine how the contribution will be carried out. For example, if someone does not have github knowledge, we may just have them send us their contribution, and we will take care of integrating the code into the repository, instead of asking for a pull request. Moreover, this year we will also welcome other non code-related contributions such as class notes etc. In essence, all of this will be decided on a per-contribution basis, so there are no general practical guidelines.

In any case, since you had this doubt, we will update the course information with a clearer explanation as soon as possible. Thank you for pointing this out.
